pkg/identity/cache/allocator.go

```diff
@@ -446,7 +447,17 @@ func (m *CachingIdentityAllocator) WaitForInitialGlobalIdentities(ctx context.Co
 		return fmt.Errorf("initial global identity sync was cancelled: %w", ctx.Err())
 	}
 
-	return m.IdentityAllocator.WaitForInitialSync(ctx)
+	if err := m.IdentityAllocator.WaitForInitialSync(ctx); err != nil {
+		return err
+	}
```

Comment:
Nit. Looks unnecessary, as already implied by the check below, but it doesn't hurt either.

Reply:
Yeah, it's only really useful for error disambiguration.